### PR TITLE
DevDocs: add doc for creating new components

### LINKED ISSUE #24066: DevDocs: Create a guide for creating new components
There is no guide for contributing new components. There is no template to use for the documentation or minimum requirements. We should add some documentation about this, or if its there, make it easier to find.

### PR DESCRIPTION
**Pull request by drw158**
This creates a checklist for people who want to create a new UI component.

Please test all links in the doc!

I want to also include a readme file requirement for adding new components, and a template example readme file.

fixes #24066

### CODE REVIEW

File: docs/new-component-checklist.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Does it fit well when placed next to or layered with other components to form a larger composition?
+
+## Functionality
+
+Do all behaviors function as expected? For responsive tabs, are menus (for overflow tabs) behaving correctly in varied device settings?
+
+## Browser Support
+
+Has the component visual quality and accuracy been assessed across Safari, Chrome, Firefox, IE and other browsers across relevant devices? Please adhere to our [browser support requirements](../README.md).
```
Comment (sixhours) [inline docs/new-component-checklist.md]: Minor tweak, first sentence in this para needs an Oxford comma. Also, is it possible to link directly to the browser support requirements using an anchor? I think you should be able to by changing the link to `README.md#browser-support`

**Comment by sixhours** [comment]
Also, will you be surfacing this new doc somewhere in the Contributing to Calypso section?

**Comment by drw158** [comment]
@sixhours Done!

**Comment by kwight** [comment]
> Not sure I meant to start a review. :)

@sixhours anyone can do a review, for sure! If the changes test well, code looks good, and tests pass, feel free to mark the PR Approved (no PR gets merged without an approval). This can be done at any time from `Files Changed > Review Changes`  👍

**Pull request merged**